src/megaapi_impl.cpp

```diff
@@ -29551,5 +29735,505 @@ int MegaTimeZoneDetailsPrivate::getDefault() const
+        json.append(",");
+    }
+
+    if (mGlobalChatsDND > -1)
+    {
+        json.append("\"CHAT\":{\"dnd\":").append(std::to_string(mGlobalChatsDND)).append("}");
+        json.append(",");
+    }
+
+    if (!mChatDND.empty() || !mChatAlwaysNotify.empty())
+    {
+        char chatid[MegaClient::CHATHANDLE * 4 / 3 + 4];
+
+        std::map<uint64_t, time_t>::const_iterator itDND;
+        for (itDND = mChatDND.begin(); itDND != mChatDND.end(); itDND++)
+        {
+            assert(isChatDndEnabled(itDND->first));
+            assert(!isChatAlwaysNotifyEnabled(itDND->first));
+
+            Base64::btoa((byte*)&(itDND->first), MegaClient::CHATHANDLE, chatid);
+            json.append("\"").append(chatid).append("\":{");
+            json.append("\"dnd\":").append(std::to_string(itDND->second)).append("}");
+            json.append(",");
+        }
+
+        std::map<uint64_t, bool>::const_iterator itAn;
+        for (itAn = mChatAlwaysNotify.begin(); itAn != mChatAlwaysNotify.end(); itAn++)
+        {
+            assert(isChatAlwaysNotifyEnabled(itAn->second));
+            assert(!isChatDndEnabled(itAn->first));
+
+            Base64::btoa((byte*)&(itAn->first), MegaClient::CHATHANDLE, chatid);
+            json.append("\"").append(chatid).append("\":{");
+            json.append("\"an\":").append("1").append("}");
+            json.append(",");
+        }
+    }
+
+    if (json.at(json.length() - 1) == ',')  // clear a tailing comma (not needed)
+    {
+        json.pop_back();
+    }
+    json.append("}");
+    return json;
+}
+
+bool MegaPushNotificationSettingsPrivate::isValid() const
+{
+    return mJsonInvalid;
+}
+
+MegaPushNotificationSettingsPrivate::~MegaPushNotificationSettingsPrivate()
+{
+
+}
+
+bool MegaPushNotificationSettingsPrivate::isGlobalEnabled() const
+{
+    return (mGlobalDND == -1 || (mGlobalDND > 0 && mGlobalDND < m_time(NULL)));
+}
+
+bool MegaPushNotificationSettingsPrivate::isGlobalDndEnabled() const
+{
+    return (mGlobalDND == 0 || mGlobalDND > m_time(NULL));
+}
+
+int64_t MegaPushNotificationSettingsPrivate::getGlobalDnd() const
+{
+    return mGlobalDND;
 }
 
+bool MegaPushNotificationSettingsPrivate::isGlobalScheduleEnabled() const
+{
+    return (mGlobalScheduleStart != -1 && mGlobalScheduleEnd != -1 && !mGlobalScheduleTimezone.empty());
+}
+
+int MegaPushNotificationSettingsPrivate::getGlobalScheduleStart() const
+{
+    return mGlobalScheduleStart;
+}
+
+int MegaPushNotificationSettingsPrivate::getGlobalScheduleEnd() const
+{
+    return mGlobalScheduleEnd;
+}
+
+const char *mega::MegaPushNotificationSettingsPrivate::getGlobalScheduleTimezone() const
+{
+    return MegaApi::strdup(mGlobalScheduleTimezone.c_str());
+}
+
+bool MegaPushNotificationSettingsPrivate::isChatEnabled(MegaHandle chatid) const
+{
+    std::map<uint64_t, time_t>::const_iterator it = mChatDND.find(chatid);
+    return (it == mChatDND.end() || it->second != 0);
```

Comment:
Should this check consider the `an` too?

Reply:
I don't think so, if chat is notifiable is managed by bool MegaApi::isChatNotifiable(MegaHandle chatid), with this function only managed if we have to enable/disable the UI
I've added a commit to modify the documentation